### DISCUSSION TITLE
tar_quarto_rep does not work on self-contained reports in sub-directories

### DISCUSSION THREAD

**Discussion by tklebel** [ropensci/tarchetypes#159] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

This is a direct continuation of #144. If we have a report in a subdirectory that has `self-contained: true`, the pipeline will succeed, but the resulting `.html` file does not include CSS and plots.

file structure:

```
_targets.R

report_dir/
├─ report.qmd
``` 

`_targets.R`:

```r
library(tarchetypes)

params <- tibble::tibble(
  variable = c("mpg", "disp"),
)

list(tar_quarto_rep(report, path = "report_dir/report.qmd", execute_params = params))
```

`report.qmd`

    ---
    title: "test"
    format: 
      html: 
        self-contained: true
    params:
      variable:
        "mpg"
    ---
 
    This report is rendered with variable = `r params$variable`.
 
    ```{r}
    library(ggplot2)
 
    ggplot(mtcars, aes(.data[[params$variable]], hp)) +
      geom_point()
    ```

Applying `self-contained: true` was necessary to circumvent the issue of the generated reports incorrectly pointing all to the same figures #157.

**Comment by wlandau** [comment]
For some reason, `self-contained: true` is having no effect. The HTML of the rendered report is surprisingly small, and I see:

```html
<p>This report is rendered with variable = mpg.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mtcars, <span class="fu">aes</span>(.data[[params<span class="sc">$</span>variable]], hp)) <span class="sc">+</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="report_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid" width="672"></p>
</div>
</div>
```

**Comment by wlandau** [comment]
So the reason the figure isn't showing up is that `targets` needs to render the report to a temporary file in the current working directory first, then afterwards move that file to its final destination.

I can reproduce what you are seeing using just Quarto:

```r
quarto::quarto_render(input = "report_dir/report.qmd", output_file = "temporary_file_in_current_working_dir.html")
```

`tar_quarto_rep()` needs to render its reports this way because of https://github.com/ropensci/tarchetypes/issues/144. If the Quarto maintainers ever relax this restriction, I will be able to move `tar_quarto_rep()` back to rendering in place.

**Reply by tklebel** [reply]
Thanks for figuring this out! Wouldn't this count as a bug on quartos side? I think it is, given the discussions in https://github.com/quarto-dev/quarto-cli/issues/5765. Hopefully this will resolve itself once the quarto maintainers fix upstream.

**Reply by wlandau** [reply]
I agree with you. Hopefully they will fix it.
